base/sparse/sparsematrix.jl

```diff
@@ -2338,9 +2338,29 @@ setindex!(A::SparseMatrixCSC, x::Number, i::Integer, J::AbstractVector{<:Integer
 setindex!(A::SparseMatrixCSC, x::Number, I::AbstractVector{<:Integer}, j::Integer) = setindex!(A, x, I, [j])
 
 # Colon translation
-setindex!(A::SparseMatrixCSC, x, ::Colon)          = setindex!(A, x, 1:length(A))
-setindex!(A::SparseMatrixCSC, x, ::Colon, ::Colon) = setindex!(A, x, 1:size(A, 1), 1:size(A,2))
-setindex!(A::SparseMatrixCSC, x, ::Colon, j::Union{Integer, AbstractVector}) = setindex!(A, x, 1:size(A, 1), j)
+function setindex!(A::SparseMatrixCSC, x, ::Colon)
+    if x == 0
+        fill!(view(A.nzval, 1:A.colptr[end]-1), 0)
+    else
+        setindex!(A, x, 1:length(A))
+    end
+end
+function setindex!(A::SparseMatrixCSC, x, ::Colon, ::Colon)
+    if x == 0
+        setindex!(A, x, :)
```

Comment:
To avoid branching twice, perhaps replicate the above `fill!` incantation here, or factor that incantation into a one-line method and call it both here and in place of the `fill!` above?